app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -202,8 +172,8 @@ class TorManager @Inject constructor(
         }
         useBridges(builtInBridges)
         if (waitForTorToStart()) return
-        LOG.info("Could not connect to Tor, stopping...")
-        stop(failedToConnect = true)
+        LOG.info("Could not connect to Tor")
+        _state.value = TorState.FailedToConnect
```

Comment:
Possible race with shutdown?

Reply:
Good point. I guess that's possible. Also, here's probably more work needed. `stop()` was removed above, so what if Tor manages to bootstrap afterwards? How does the UI handle this? I guess if we give up at this point, we want to stop Tor.  But so far we are only setting `ShareUiState.Error`, but then we let things run.

Reply:
This race still exists in the new version, but I wonder what the problem could be. If the user stopped the sharing and then we run into this, showing either stopped or failed to connect won't hurt. Or do you think of something else here?

Reply:
Good point, it's always possible for the timeout to happen a millisecond before the user taps the stop button, in which case the UI will go to the error state rather than the stopping state. And like you say, that won't hurt.

What about the case where the user already tapped the stop button a while ago, and the UI went to the stopping state? In that case I think we'd prefer it if the UI ended up in the stopped state, not the error state.

Previously the `TorState` would have moved to `Stopping` at roughly the same time as the UI entered the stopping state, so in the case where the UI was already in the stopping state for a while, we would expect to have returned from one of the `waitForTorToStart()` checks above and not have reached this line. But now that `TorState.Stopping` no longer exists, I think we wouldn't return from `waitForTorToStart()` until we reached `TorState.Stopped`, so it seems to me that there is now a risk that `TorState.Stopped` would get overwritten by `TorState.FailedToConnect`. Does that make sense?

Reply:
So are you advocating for bringing back the `Stopping`  state or introducing a new solution with `AtomicReference` and check here if we are already stopped?

Reply:
Let's check whether my guess above is correct, and we don't return from `waitForTorToStart()` until we reach `TorState.Stopped`. If that's correct then I think we may need to reintroduce the `Stopping` state as a way of telling `waitForTorToStart()` to quit, so that we don't overwrite `Stopped` with `FailedToConnect`. (In any case, it's probably sensible that we don't keep trying to change the connection method after we've told Tor to stop.)

Reply:
If reintroducing `Stopping` would cause other problems, we can look for another way to tell `waitForTorToStart()` to quit. But since it's already monitoring the `TorState`, reintroducing `Stopping` seems like it might be a natural solution?

Reply:
OK, I tested this and there's no need to reintroduce `Stopping` for this purpose. `waitForTorToStart()` is killed with a `JobCancellationException` when `startCheckJob` is cancelled, so it doesn't live long enough to overwrite `Stopped` with `FailedToConnect`.